src/app/EventManagement.h

```diff
@@ -360,7 +362,39 @@ class EventManagement
      */
     void SetScheduledEventInfo(EventNumber & aEventNumber, uint32_t & aInitialWrittenEventBytes);
 
+    enum class BypassACL : uint8_t
+    {
+        kNoBypass   = 0,
+        kAlwaysPass = 1,
+        kAlwaysFail = 2,
+    };
+
+    void SetBypassACL(BypassACL aConfig) { mBypassACL = aConfig; }
```

Comment:
Is this only needed for tests?  We should not be shipping this in release software; it increases attack surface...

Reply:
agree, don't put it in release software, consider to use CONFIG_IM_BUILD_FOR_UNIT_TEST?

Reply:
Please coordinate with me about Access Control during testing. Yes, it's a concern I'm trying to keep under control. Don't add bypass flags or compilation flags.

Instead, in a test access control can be bypassed by configuring a different AccessControl::Delegate (implementation), such as the new "permissive" one. Alternatively, in test code an "admin for CASE subject" ACL can be installed, then even other event listeners (not just the commissioner or admin) can succeed in IM operations during tests. There's issues here with leaving the ACL in a good state, so other tests can succeed (e.g. on devices where all tests run in one app).

I know this isn't ideal, but those are the approaches in the short term, until I can get cycles to improve it more (which will eventually rise to the top of my priority list, but isn't now).

Reply:
Just looking at this PR in more detail, it seems two unit test files were affected. I did add code in AppTestContext.cpp which sets the "permissive" access control implementation, so these app level tests (TestCommandInteraction, for example) should pass. So you shouldn't need another bypass. If you do, please let me know what is failing/complaining, so I can address it or at least provide a recommendation for how to address it. Thanks!

Reply:
Commented out the test with a TODO for #10253